File: StudyBoard.Auth.WebApi/Model/RegistrationRequest.cs
using System.Collections.Generic;

namespace StudyBoard.Auth.WebApi.Model {
	public class RegistrationRequest {
		public string FirstName { get; set; }
		public string LastName { get; set; }
		public string PhoneNumber { get; set; }
		public string Email { get; set; }
		public string Login { get; set; }
		public string Password { get; set; }
		public List<int> Roles { get; set; }
	}
}

File: study-board-web/src/app/core/model/auth/registration-request.ts
import { Registration } from "./registration";

export class RegistrationRequest {
	public name: string;
	public login: string;
	public password: string;
	public firstName: string;
	public lastName: string;
	public phoneNumber: string;
	public email: string;

	constructor(registration: Registration) {
		this.name = registration.name;
		this.login = registration.login;
		this.password = registration.password;
		this.firstName = registration.firstName;
		this.lastName = registration.lastName;
		this.phoneNumber = registration.phoneNumber;
		this.email = registration.email;
	}
}